src/main/java/org/spongepowered/api/service/profile/GameProfileCache.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.service.profile;
+
+import java.util.Collection;
+import java.util.UUID;
+
+import org.spongepowered.api.GameProfile;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+/**
+ * A service which provides access to the local {@link GameProfile} cache.
+ *
+ * <p>Note that the stored user name may not always be up to date.</p>
+ */
+public interface GameProfileCache extends GameProfileSource {
+
+    /**
+     * Returns a collection of all cached profiles.
+     *
+     * <p>This collection may include profiles of players who never joined the
+     * server.</p>
+     *
+     * @return A collection of {@link GameProfile}s
+     */
+    ListenableFuture<Collection<GameProfile>> getAll();
```

Comment:
Why is this a future? The cached game profiles should be available without additional query time